### ISSUE TITLE
New tabs don't become children of the current tab, if the tab is opened from links in subframes

### ISSUE THREAD

**Issue by piroor** [piroor/treestyletab#1174]
## Short description

In some cases, tabs automatically opened from links or something are not associated to the current tab.

## Steps to reproduce

 1. Start Filrefox with clean profile.
 2. Install TST.
 3. Open https://twitter.com/piro_or/status/763756236751851521 in a tab.
 4. Click the twitter card for the linked contents.

## Expected result

A new child tab is opened.

## Actual result

An independent tab is opened.

## Environment

 * Platform (OS): Windows 7 (64bit)
 * Version of Firefox: 48.0
 * Version (or revision) of Tree Style Tab: master, HEDA 82016.8.12)

**Comment by piroor** [comment]
logs in the browser console, with `extensions.treestyletab.debug.base`=`true`:

~~~
[treestyletab:base] new orphan tab is requested.  utils_logWithStackTrace@resource://treestyletab-modules/utils.js:296:16
  logWithStackTrace@resource://treestyletab-modules/base.js:89:2
  TSTBase_readyToOpenOrphanTab@resource://treestyletab-modules/base.js:1384:3
  TSTBase_readyToOpenOrphanTabNow@resource://treestyletab-modules/base.js:1397:7
  TSTWindow_onBeforeBrowserAccessOpenURI@resource://treestyletab-modules/window.js:1452:4
  TSTWH_preInit/nsBrowserAccess.prototype.openURIInFrame@chrome://treestyletab/content/windowHelper.js:41:5
~~~

**Comment by piroor** [comment]
More logs with `extensions.treestyletab.debug.base`=`true`:

~~~
[treestyletab:window] TSTWindow_onBeforeBrowserAccessOpenURI: opener is nsIOpenURIInFrameParams
[treestyletab:window]   opener =>[null, false]
[treestyletab:base] new orphan tab is requested.  utils_logWithStackTrace@resource://treestyletab-modules/utils.js:296:16
  logWithStackTrace@resource://treestyletab-modules/base.js:89:2
  TSTBase_readyToOpenOrphanTab@resource://treestyletab-modules/base.js:1384:3
  TSTBase_readyToOpenOrphanTabNow@resource://treestyletab-modules/base.js:1397:7
  TSTWindow_onBeforeBrowserAccessOpenURI@resource://treestyletab-modules/window.js:1455:4
  TSTWH_preInit/nsBrowserAccess.prototype.openURIInFrame@chrome://treestyletab/content/windowHelper.js:41:5
~~~

**Comment by piroor** [comment]
Because the twitter card is an inline frame, its "referrer" is different from the tab's top level URL. Hmm...

**Comment by piroor** [comment]
This problem should be fixed by these commits:
259bdf7c86fe9b4f58d73f7a63e197a0f9d59ef8
1a42894f3ee1f4b8e022dcd943d05314e32cc99e
4e4ec57324f9a81db6728eb9c601b557e6412fc6
fee1428f5459f72a19b27f2f6c9ca45daa63a2ac

**Comment by piroor** [comment]
I've released a new version 0.17.2016083001 includes commits for this issue, thus I close this.
